pkg/apis/apps/v1alpha1/servicebindingrequest_types.go

```diff
@@ -81,6 +81,73 @@ type ApplicationSelector struct {
 	LabelSelector               *metav1.LabelSelector `json:"labelSelector,omitempty"`
 	metav1.GroupVersionResource `json:",inline"`
 	ResourceRef                 string `json:"resourceRef,omitempty"`
+
+	// BindingPath refers to the path in the application workload's schema
+	// where the binding workload would be referenced.
+	// +optional
+	BindingPath *BindingPath `json:"bindingPath,omitempty"`
+}
+
+const (
+	// DefaultContainersPath has the logical path logical path
+	// to find containers on supported objects
+	// Used as []string{"spec", "template", "spec", "containers"}
+	DefaultPathToContainers = "spec.template.spec.containers"
+
+	// DefaultPathToVolumes is the logical path to find volumes on supported objects
+	// used as []string{"spec", "template", "spec", "volumes"}
+	DefaultPathToVolumes = "spec.template.spec.volumes"
+)
+
+// SetDefaults set default value for binding path
+func (applicationSelector *ApplicationSelector) SetDefaults() {
+	if applicationSelector.LabelSelector == nil {
+		applicationSelector.LabelSelector = &metav1.LabelSelector{}
+	}
+	if applicationSelector.BindingPath == nil {
+		applicationSelector.BindingPath = &BindingPath{
+			PodSpecPath: &PodSpecPath{
+				Containers: DefaultPathToContainers,
+				Volumes:    DefaultPathToVolumes,
+			},
+		}
+	} else {
+		if applicationSelector.BindingPath.PodSpecPath == nil {
+			applicationSelector.BindingPath = &BindingPath{
```

Comment:
I believe we do not want to override `BindingPath` in the case `BindingPath.PodSpecPath` is `nil`; in this case, a `PodSpecPath` value should be created and assigned to `applicationSelector.BindingPath.PodSpecPath`.

Additionally, being `PodSpecPath` not nil, its `Containers` and `Volumes` fields should be queried to verify whether it is not empty and have the defaults applied in the case they are.

Reply:
Done. 7f036ded64abb5ffff9740569900e2a4dbc0de84